src/frontend/sharedComponents/Editor/index.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import {Photo} from '../../contexts/PhotoPromiseContext/types';
+import {DescriptionField} from './DescriptionField';
+import {MediaScrollView} from '../MediaScrollView';
+import {PresetAndLocationView} from './PresetAndLocationView';
+import {ScreenContentWithDock} from '../ScreenContentWithDock';
+import {ActionButtons} from './ActionButtons';
+
+type EditorProps = {
+  presetName: string;
+  onPressPreset?: () => void;
+  PresetIcon: React.ReactNode;
+  notes: string;
+  updateNotes: (newNotes: string) => void;
+  photos: (Partial<Photo> | undefined)[];
+  location?: {
+    lat: number | undefined;
+    lon: number | undefined;
+    accuracy: number | undefined;
+  };
+  actionButtonChildren?: React.ReactNode;
+  fieldIds?: string[];
+};
+
+export const Editor = ({
+  presetName,
+  onPressPreset,
+  PresetIcon,
+  notes,
+  updateNotes,
+  photos,
+  location,
+  fieldIds,
+  actionButtonChildren,
+}: EditorProps) => {
+  return (
+    <ScreenContentWithDock
+      dockContainerStyle={{padding: 0}}
+      dockContent={
+        actionButtonChildren || <ActionButtons fieldIds={fieldIds} />
```

Comment:
not keen on this interface. the `fieldIds` prop being conditionally applied based on another prop's existence is a smell to me. i would suggest removing the `fieldIds` prop and then renaming `actionButtonChildren` to something like `actionsRow` and making it a required prop:

```tsx
export const Editor = ({
  presetName,
  onPressPreset,
  PresetIcon,
  notes,
  updateNotes,
  photos,
  location,
  actionsRow,
}: EditorProps) => {
  return (
    <ScreenContentWithDock
      dockContainerStyle={{padding: 0}}
      dockContent={actionsRow}>
      <PresetAndLocationView
        onPressPreset={onPressPreset}
        presetName={presetName}
        PresetIcon={PresetIcon}
        location={location}
      />
      <DescriptionField notes={notes} updateNotes={updateNotes} />
      <MediaScrollView photos={photos} />
    </ScreenContentWithDock>
  );
};
```

and then the usage of it looks like this:

```tsx
<Editor actionsRow={<ActionButtons fieldIds={fieldIds} />} />
```

Reply:
on a related note, although perhaps outside the desired scope of this PR, the `ActionTab` component (which existed prior to this) feels like an overabstraction, which is causing some of what's being introduced here to be more complicated than necessary.

probably more changes that you'd like to implement, but I'd prefer the usage of the Editor to ultimately look more like this: 

```tsx
<Editor
  actionsRow={
    <EditorActions>
      <CameraActionButton />
      <DetailsActionButton />
    </EditorActions>
  }
/>
```

Reply:
I think the abstraction actually makes sense because of the keyboard view. We customize whats on the top part of the keyboard based on what actions are available. I think if the keyboard didn't change I would agree though!

Reply:
https://github.com/digidem/comapeo-mobile/pull/416/commits/d4dbe937cb9adcb6e6cabb36dde7d11bf613c683